src/distribution/pert.rs

```diff
@@ -0,0 +1,339 @@
+            0.4938271604938269,
+            0.6249999999999999,
+            0.49382716049382713,
+            0.1929012345679011,
+            0.022562499999999933,
+            0.0,
+        ];
+        assert::close(
+            &x.iter().map(|&x| d.density(x)).collect::<Vec<_>>(),
+            &x.iter().map(|&x| beta.density(x)).collect::<Vec<_>>(),
+            1e-14,
+        );
+        assert::close(
+            &x.iter().map(|&x| d.density(x)).collect::<Vec<_>>(),
+            &p,
+            1e-14,
+        );
+    }
+
+    #[test]
+    fn distribution() {
+        let d = new!(-1.0, 0.5, 2.0);
+        // let d = new!(0.0, 0.5, 1.0);
+        let beta = Beta::new(3.0, 3.0, -1.0, 2.0);
+        let x = vec![-1.15, -1.0, -0.85, -0.5, 0.0, 0.5, 1.0, 1.5, 1.85, 2.0];
+        let p = vec![
+            0.0,
+            0.0,
+            0.001158125,
+            0.03549382716049382,
+            0.20987654320987656,
+            0.5,
+            0.7901234567901234,
+            0.9645061728395061,
+            0.998841875,
+            1.0,
+        ];
+        assert::close(
+            &x.iter().map(|&x| d.distribution(x)).collect::<Vec<_>>(),
+            &x.iter().map(|&x| beta.distribution(x)).collect::<Vec<_>>(),
+            1e-14,
+        );
+        assert::close(
+            &x.iter().map(|&x| d.distribution(x)).collect::<Vec<_>>(),
+            &p,
+            1e-14,
+        );
+    }
+
+    #[test]
+    fn entropy() {
+        use std::f64::consts::E;
+        let d = vec![
+            new!(0.0, 0.5, 1.0),
+            new!(0.0, 0.5, E),
+            new!(0.0, 0.3, 1.0),
+            new!(-1.0, 1.0, 2.0),
+        ];
+        assert::close(
+            &d.iter().map(|d| d.entropy()).collect::<Vec<_>>(),
+            &d.iter()
+                .map(|d| Beta::new(d.alpha(), d.beta(), d.a(), d.c()).entropy())
+                .collect::<Vec<_>>(),
+            1e-15,
+        );
+    }
+
+    #[test]
+    fn inverse() {
+        let d = new!(-1.0, 0.5, 2.0);
+        let p = vec![0.0, 0.2, 0.4, 0.6, 0.8, 1.0];
+        let x = vec![
+            -1.,
+            -0.020206186475766774,
+            0.33876229245942,
+            0.6612377075405802,
+            1.0202061864757672,
+            2.0,
+        ];
+        assert::close(
+            &p.iter().map(|&p| d.inverse(p)).collect::<Vec<_>>(),
+            &x,
+            1e-14,
+        );
+    }
+
+    #[test]
+    fn kurtosis() {
+        assert_eq!(new!(0.0, 0.5, 1.0).kurtosis(), -2.0 / 3.0);
+    }
+
+    #[test]
+    fn mean() {
+        assert_eq!(new!(0.0, 0.5, 1.0).mean(), 0.5);
+        assert_eq!(new!(-1.0, 1.5, 2.0).mean(), (1.5 * 4.0 - 1.0 + 2.0) / 6.0);
+        assert_eq!(
+            Beta::new(3.0, 3.0, -1.0, 2.0).mean(),
+            (0.5 * 4.0 - 1.0 + 2.0) / 6.0
+        );
+    }
+
+    #[test]
+    fn median() {
+        assert_eq!(new!(0.0, 0.5, 1.0).median(), 0.5);
+        // XXX sandbox.open.wolframcloud.com gives 0.350999???
```

Comment:
Which one is correct? Can you try computing it elsewhere? Wikipedia says that there is also a modified version; can Wolfram use that one by default?

Reply:
scipy seems to agree with wolfram:

```
>>> from scipy.stats import beta
>>> x = beta(2.2, 3.8)
>>> x.median()
0.3509994849491181
```

after look at wikipedia, the Beta median() doesn't look correct - it uses the approximate formula instead of the inverse regularized beta function.